apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,103 @@
+import { RequestHandler } from 'express';
+import Post from '../models/post';
+import Comment from '../models/comment';
+
+/**
+ Get all posts 
+ */
+export const getPosts: RequestHandler = async (_req, res) => {
+  try {
+    const posts = await Post.find();
+    return res.status(200).json(posts);
+  } catch (error) {
+    const { message } = error;
+    return res.status(500).json({ message });
+  }
+};
+
+/**
+ * Create a new post
+ */
+export const createPost: RequestHandler = async (req, res) => {
+  const postData = req.body;
+  try {
+    const newPost = await Post.create(postData);
+    return res.status(201).json(newPost);
+  } catch (error) {
+    const { message } = error;
+    return res.status(500).json({ message });
+  }
+};
+
+/**
+ * Update a post
+ */
+export const patchPost: RequestHandler = async (req, res) => {
+  const { id } = req.params;
+  const newPostData = req.body;
+  try {
+    const modifiedPost = await Post.findByIdAndUpdate(id, newPostData, { new: true, runValidators: true });
```

Comment:
This method will return a falsy value if the post doesn't exist, for better error handling we can validate if modifiedPost exist if not return a 404.